x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -404,8 +404,23 @@ public void usageStats(final ActionListener<Map<String, Object>> listener) {
 
     @Override
     public void reload(Settings settings) {
-        var clientSecret = CLIENT_AUTHENTICATION_SHARED_SECRET.getConcreteSettingForNamespace(this.realmRef().getName()).get(settings);
-        this.clientAuthenticationSharedSecret.rotate(clientSecret, config.getSetting(CLIENT_AUTH_SHARED_SECRET_ROTATION_GRACE_PERIOD));
+        final SecureString newClientSharedSecret = CLIENT_AUTHENTICATION_SHARED_SECRET.getConcreteSettingForNamespace(
+            this.realmRef().getName()
+        ).get(settings);
+
+        // Validate settings' consistency before rotating the secret.
```

Comment:
nit: appreciate the comments, but this one probably is not necessary (doing validation is common and is mostly evident by the name)